tests/unit/fem/test_linearform_ext.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "unit_tests.hpp"
+#include <functional>
+
+using namespace mfem;
+
+struct LinearFormExtTest
+{
+   enum { DomainLF, DomainLFGrad, VectorDomainLF, VectorDomainLFGrad };
+   using vector_fct_t = void(const Vector&, Vector&);
+   const double abs_tol = 1e-14, rel_tol = 1e-14;
+   const char *mesh_filename;
+   Mesh mesh;
+   const int dim, vdim, ordering, gll, problem, p, q, SEED = 0x100001b3;
+   H1_FECollection fec;
+   FiniteElementSpace vfes, mfes;
+   const Geometry::Type geom_type;
+   IntegrationRules IntRulesGLL;
+   const IntegrationRule *irGLL, *ir;
+   Array<int> elem_marker;
+   Vector one_vec, dim_vec, vdim_vec, vdim_dim_vec;
+   ConstantCoefficient cst_coeff;
+   VectorConstantCoefficient dim_cst_coeff, vdim_cst_coeff, vdim_dim_cst_coeff;
+   std::function<vector_fct_t> vdim_vec_function =
+   [&](const Vector&, Vector &y) { y.SetSize(vdim); y.Randomize(SEED); };
```

Comment:
This is pretty much equivalent to a `VectorConstantCoefficient` since the `y` vector will always be the same. One possibility to make a true variable coefficient would be to have the random seed depend on the input vector.

Reply:
Changed to `y.Randomize(static_cast<int>(std::time(NULL)))`

Reply:
Reverted, add `QuadratureLFIntegrator` and `VectorQuadratureLFIntegrator` tests.